web/wp-content/plugins/tmbi-video/inc/class-video-cpt-setting.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * Class Video_CPT_Setting
+ *
+ * Helper functions for Video CPT
+ */
+class Video_CPT_Setting {
+
+	/**
+	 * Video_CPT_Setting constructor.
+	 */
+	public function __construct() {
+		add_action( 'acf/init', array( $this, 'register_acf_video_fields' ) );
```

Comment:
Using `array( $this, ... )` for the callback breaks the `remove_action` API. If not going to address with this PR, please create a tech debt ticket and link it here!

Reply:
I think that's not a tech dept. because without that hook we cannot have video CPT fields. when we need to prioritize this hook, we can always change this hook's priority. If we need to use the remove_action, we need to use static functions. Having static functions for everything won't help much on OOP approach.

Reply:
>  without that hook we cannot have video CPT fields
This hook is for a specific implementation approach (using ACF) to meet a product requirement.  Either of those can change for different reasons. Breaking the `remove_action` API limits our ability to scale the code safely and quickly.

> If we need to use the remove_action, we need to use static functions. 
Namespaced functions can work just as fine for the purpose of code organization as there's no class-level mutable state.

> Having static functions for everything won't help much on OOP approach.
I don't see what other benefits of OOP (other than namespacing) we're using here. What's the reason to use this pattern? What's the benefit of it that's worth breaking a core WordPress API?

Reply:
exactly. that's what I'm saying. it's a requirement of ACF. if we remove it, it would remove functionality of Video CPT. Personally I would remove the dependency of ACF. 
its not breaking the WordPress core removal functionality. We need to be more careful when we add/remove code from front end. these hooks are only for admin. 

not in this example. but if I have to manage local variable, having static variables don't help. static will be used whenever it needed.

if you think it will be a tech-dept. please create ticket for that. thanks.

If you think we should follow Spaghetti Code, please document the structure, we will follow the same.

Reply:
Breaking a core WP API _is_ a problem. It's not for this hook only, it's for all hooks we add: we shouldn't break things.

This an **engineering** project with four main goals. So what we're doing here is to do things in **the best way possible** when measured against those goals. This is not an artistic project and we shouldn't be guided by personal, stylistic choices. Whether we use spaghetti code, OOP, functional, it is only relevant when measured for speed, stability, scalability or maintainability. 

If our code breaks an API, that goes against the project goals and it should be fixed _unless_ there's a strong reason to do it that way. Breaking things is bad, and only in rare occasions something that's bad is the only way to do it (let alone the _best_ way to do it). And again, the goal of the project is to port over functionality **in the best way possible**, not to copy, paste and reformat legacy code.

In any case, I think this issue and others in those lines can be solved easily. I'll make sure to document the quality standards if there's something else to clarify.

Thanks!